protocol/msgpack/options.go

```diff
@@ -20,111 +20,139 @@
 
 package msgpack
 
-import (
-	"github.com/m3db/m3metrics/policy"
-	xpool "github.com/m3db/m3x/pool"
-)
+import xpool "github.com/m3db/m3x/pool"
 
 const (
 	// Whether the iterator should ignore higher-than-supported version
-	// by default for unaggregated metrics
+	// by default for unaggregated iterator.
 	defaultUnaggregatedIgnoreHigherVersion = false
 
+	// Default reader buffer size for the unaggregated iterator.
+	defaultUnaggregatedReaderBufferSize = 1440
+
+	// Whether a float slice is considered a "large" slice and therefore
+	// resort to the pool for allocating that slice.
+	defaultLargeFloatsSize = 1024
+
 	// Whether the iterator should ignore higher-than-supported version
-	// by default for aggregated metrics
+	// by default for aggregated iterator.
 	defaultAggregatedIgnoreHigherVersion = false
+
+	// Default reader buffer size for the aggregated iterator.
+	defaultAggregatedReaderBufferSize = 1440
 )
 
 type unaggregatedIteratorOptions struct {
 	ignoreHigherVersion bool
-	floatsPool          xpool.FloatsPool
-	policiesPool        policy.PoliciesPool
+	readerBufferSize    int
+	largeFloatsSize     int
+	largeFloatsPool     xpool.FloatsPool
 	iteratorPool        UnaggregatedIteratorPool
 }
 
 // NewUnaggregatedIteratorOptions creates a new set of unaggregated iterator options
```

Comment:
nit: missed a period here. We might want to do a quick audit of the codebase when we have the time so we can standardize the formatting a little more. On a side note, it would be cool if `gofmt` had configuration options so that you can specify things like these that it doesn't have an opinion on.